packages/elements/src/components/MarkdownViewer/CustomComponents/CodeComponent.tsx

```diff
@@ -0,0 +1,73 @@
+import { ICode } from '@stoplight/markdown/ast-types/smdast';
+import { defaultComponentMapping, ICodeAnnotations, IComponentMappingProps } from '@stoplight/markdown-viewer';
+import { HttpParamStyles, IHttpOperation, IHttpRequest } from '@stoplight/types';
+import { get, isObject } from 'lodash';
+import React from 'react';
+import URI from 'urijs';
+
+import { useParsedValue } from '../../../hooks/useParsedValue';
+import { isHttpOperation, isHttpRequest, isJSONSchema } from '../../../utils/guards';
+import { SchemaAndDescription } from '../../SchemaAndDescription';
+import { TryIt } from '../../TryIt';
+
+export const CodeComponent = (props: IComponentMappingProps<ICode<ICodeAnnotations>>) => {
+  const {
+    node: { annotations, value, resolved, meta },
+  } = props;
+  const nodeType = get(annotations, 'type') || meta;
+
+  const parsedValue = useParsedValue(resolved ?? value);
+
+  if (nodeType === 'json_schema') {
+    if (!isJSONSchema(parsedValue)) {
+      return null;
+    }
+
+    return <SchemaAndDescription title={annotations?.title} schema={parsedValue} />;
+  }
+
+  if (nodeType === 'http') {
+    if (!isObject(parsedValue) || (!isHttpRequest(parsedValue) && !isHttpOperation(parsedValue))) {
+      return null;
+    }
+
+    return <TryIt httpOperation={isHttpOperation(parsedValue) ? parsedValue : parseHttpRequest(parsedValue)} />;
+  }
+
+  const DefaultCode = defaultComponentMapping.code!;
+  return <DefaultCode {...props} />;
+};
+
+export function parseHttpRequest(data: IHttpRequest): IHttpOperation {
+  const uri = URI(data.url);
+  return {
+    id: '?http-operation-id?',
+    method: data.method,
+    path: uri.is('absolute') ? uri.path() : data.url,
+    servers: [{ url: uri.is('absolute') ? uri.origin() : data.baseUrl }],
+    request: {
+      ...(data.query && Object.keys(data.query).length
+        ? {
+            query: Object.entries(data.query).map(([key, value]) => ({
+              name: key,
+              style: HttpParamStyles.Form,
+              schema: { default: Array.isArray(value) && value.length > 0 ? value[0] : value },
+            })),
+          }
+        : null),
```

Comment:
```suggestion
query: Object.entries(data.query).map(([key, value]) => ({
        name: key,
        style: HttpParamStyles.Form,
        schema: { default: Array.isArray(value) && value.length > 0 ? value[0] : value },
      })),
```

Same logic applies to `headers`.

Now I know this is controversial a littlebit, as I've removed two of your checks.
1. `Object.keys(data.query).length`. I don't think this is needed as TryIt should work just fine with an empty array passed in (IIRC we default the `undefined`s to an empty array anyway.)
2. `data.query` should always be truthy per TypeScript. I know this may not be the case, but then let's fix those TS types instead. I think the core problem is back in `CodeComponent`. You use the type `IHttpRequest` but really, you are supporting `Pick<IHttpRequest, 'method' | 'url'> & Partial<IHttpRequest>`, aren't you? You can either propagate this change throughout your changes, or provide defaults for `baseUrl`, `headers` and `query` right in `CodeComponent` (which may be simpler). But the current type-guard approach is way too weak IMO and creates a lie in many real world use cases.

Reply:
1. I wanted to avoid adding that empty array but there's no problem with having it there anyway so I removed it.
2. That's correct. I've improved those types and the guard itself so it wouldn't break in case of some weird input.

Reply:
Nice! Thank you!